packages/babel-traverse/test/arrow-transform.js

```diff
@@ -0,0 +1,439 @@
+import { NodePath } from "../lib";
+import assert from "assert";
+import { parse } from "babylon";
+import generate from "babel-generator";
+import * as t from "babel-types";
+
+function assertConversion(input, output, {
+  methodName = "method",
+  extend = false,
+  arrowOpts,
+} = {}) {
+  const inputAst = wrapMethod(input, methodName, extend);
+  const outputAst = wrapMethod(output, methodName, extend);
+
+  const rootPath = NodePath.get({
+    hub: {
+      file: {
+        addHelper(helperName) {
+          return t.memberExpression(t.identifier("babelHelpers"), t.identifier(helperName));
+        },
+      },
+    },
+    parentPath: null,
+    parent: inputAst,
+    container: inputAst,
+    key: "program",
+  }).setContext();
+
+  rootPath.traverse({
+    ClassMethod(path) {
+      path.get("body.body.0.expression").arrowFunctionToExpression(arrowOpts);
+    },
+  });
+
+  assert.equal(generate(inputAst).code, generate(outputAst).code);
+}
+
+function wrapMethod(body, methodName, extend) {
+  return parse(`
+    class Example ${extend ? ("extends class {}") : ""} {
+      ${methodName}() {${body} }
+    }
+  `);
+}
+
+describe("arrow function conversion", () => {
+  it("should convert super calls in constructors", () => {
+    assertConversion(`
+      () => {
+        super(345);
+      };
+      super();
+      () => super();
+    `, `
+      var _supercall = (..._args) => super(..._args);
+
+      (function () {
+        _supercall(345);
+      });
+      _supercall();
+      () => _supercall();
+    `, { methodName: "constructor" });
+  });
+
+  it("should convert super calls and this references in constructors", () => {
+    assertConversion(`
+      () => {
+        super(345);
+        this;
+      };
+      super();
+      this;
+      () => super();
+      () => this;
+    `, `
+      var _supercall = (..._args) => _this = super(..._args),
```

Comment:
Was wondering if something would catch multiple `super()` invocations, but I guess that's the job of the JS engine / class transform.